flink-runtime-web/web-dashboard/src/app/interfaces/apis.ts

```diff
@@ -0,0 +1,159 @@
+import { HttpEvent } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import {
+  Checkpoint,
+  CheckpointConfig,
+  CheckpointDetail,
+  CheckpointSubTask,
+  JobAccumulators,
+  JobBackpressure,
+  JobConfig,
+  JobDetailCorrect,
+  JobException,
+  JobFlameGraph,
+  JobManagerLogItem,
+  JobsItem,
+  JobSubTask,
+  JobSubTaskTime,
+  JobVertexTaskManager,
+  JobMetric,
+  MetricMap,
+  MetricMapWithTimestamp,
+  Overview,
+  TaskManagerDetail,
+  TaskManagerLogItem,
+  TaskManagerLogDetail,
+  TaskManagersItem,
+  Watermarks,
+  JobManagerLogDetail,
+  JobManagerConfig,
+  JarList,
+  PlanDetail
+} from 'interfaces';
+
+export interface APIService {
+  getUrlPrefix(): string;
+}
+
+export abstract class OverviewAPIService implements APIService {
```

Comment:
What's the purpose of extracting these base classes? They all just have one implementation. They currently just duplicate the structure (and unnecessarily so since TypeScript is structurally typed anyway), and do so in a location far away from the actual implementation.